Revamp README.Md

bettio opened the pull request:
Update it, AtomVM is less in the early stage, and more solid nowadays.

These changes are made under both the "Apache 2.0" and the "GNU Lesser General
Public License 2.1 or later" license terms (dual license).

SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later

UncleGrumpy commented on README.Md:
```diff
@@ -91,23 +91,43 @@ Project Status
+- The standard library is minimal — some functions and modules are not yet implemented.
+- Features that depend on a full operating system (e.g. file I/O, OS processes) may be missing or
+behave differently.
+- Certain BEAM features like module `on_load` functions
```
It looks like this trails off mid sentence.

UncleGrumpy reviewed (commented):
This looks great, there is just one place that looks like a point is cut-off mid sentence.

UncleGrumpy reviewed (approved):
This looks perfect.

UncleGrumpy commented:
The one other place outside the README that we probably want to update at the same time is the “warning” on the landing page of the published docs:
https://github.com/atomvm/AtomVM/blob/0578eb49fab938330e157b64557440022eb67049/doc/src/index.rst?plain=1#L19

this should probably be changed to a “notice” to the effect that AtomVM is in a beta stage overall and that APIs are likely to change from one release to the next until a stable 1.0 release.

bettio commented:
> The one other place outside the README that we probably want to update at the same time is the “warning” on the landing page of the published docs:
> 
> https://github.com/atomvm/AtomVM/blob/0578eb49fab938330e157b64557440022eb67049/doc/src/index.rst?plain=1#L19
> 
> this should probably be changed to a “notice” to the effect that AtomVM is in a beta stage overall and that APIs (especially peripheral drivers) may change from one release to the next until a stable 1.0 release.

I hope the change I did addresses your comment.

UncleGrumpy reviewed (approved):
That was the right place, this looks great.
